[Pontakorn Prasertsuk] iP

peppapighs opened the pull request:
### Bob Chatbot ###

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Bob Chatbot helps you remember your tasks. It's,
- text-based
- easy to learn/use
- blazingly fast!

To use, simply following these steps
1. Download the jar file from [here](https://github.com/peppapighs/ip/releases/tag/A-CodingStandard)
2. Double click the jar file to open the program
3. Add your tasks
4. Let the bot manage your tasks!

and it's **FREE**

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

peppapighs committed:
ignore .class file

peppapighs committed:
Level-1: greeting text & reading commands

peppapighs committed:
read input as line

peppapighs committed:
Level-2: add items to the list

peppapighs committed:
add task class with status

peppapighs committed:
Level-3: mark task as done and not done

peppapighs committed:
split input into arguments

peppapighs committed:
add todo & deadline tasks

peppapighs committed:
Level-4: add event task

peppapighs committed:
A-TextUiTesting: add test case

peppapighs committed:
Level-5: handle possible errors

peppapighs committed:
Level-6: delete task

peppapighs committed:
Format code with google-java-style

peppapighs committed:
Refactor bot main logic

peppapighs committed:
Use regex for splitting command

peppapighs committed:
Add save & load tasks from file

peppapighs committed:
Format date with LocalDate

peppapighs committed:
Merge branch 'branch-Level-8'

peppapighs committed:
Extract & package classes

peppapighs committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

peppapighs committed:
Add Command classes for executing commands

peppapighs committed:
Refactor try-catch in Parser

peppapighs committed:
Rename exception package to common package

peppapighs committed:
Add encode test for Task

peppapighs committed:
Fix task decode bug

peppapighs committed:
Show numeric order for task list

peppapighs committed:
Add TaskList & Parser tests

peppapighs committed:
Configure gradle main class name

peppapighs committed:
Add Javadoc

peppapighs committed:
Add find command

peppapighs committed:
Merge branch 'branch-Level-9'

peppapighs committed:
Merge branch 'branch-A-JavaDoc'

peppapighs committed:
Add Javadoc for FindCommand

peppapighs committed:
Format code to follow coding standard

peppapighs committed:
Reformat the coding style

peppapighs committed:
Add GUI for chatbot

peppapighs committed:
Merge branch 'branch-Level-10'

peppapighs committed:
Add JavaDoc for GUI classes

angkl0 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,77 @@
+     * @param title the name of the task
+     * @param status whether the task is completed or not
+     */
+    Task(String title, boolean status) {
```
Could there be an access modifier for this method?

peppapighs replied:
I have include `public` as the access modifier. Thanks for pointing this out.

angkl0 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,77 @@
+     * @param status the status to be set
+     * @return the task after changing status
+     */
+    Task setStatus(boolean status) {
```
Same as above.

angkl0 reviewed (commented):
Overall, code standard is high. There could be more consistency as to whether there is an empty line before return statement. The ones in tasklist class do not while most in other classes do.

peppapighs committed:
Add access modifiers for Task

yunruu reviewed (commented):
Great code, clean and elegant!

peppapighs committed:
Increase consistency with blank line before return

Some return statements have no blank line before them.

Having blank line before return statements increases consistency in code quality.

peppapighs committed:
Merge pull request #1 from peppapighs/branch-A-CodeQuality

Increase consistency with blank line before return

peppapighs committed:
Add blank line before return statements

There are missing blank lines in TaskList and Duke.

To increase code quality consistency.

peppapighs committed:
Add assertion for file creation

File should be created when interacting with Storage.

Assertion helps with verifying assumption that should be true.

Java assertion is for verifying assumption during runtime and can be turned off.

peppapighs committed:
Merge pull request #2 from peppapighs/branch-A-Assertions

Add assertion for file creation

peppapighs committed:
Add gradle CI

Currently no workflow when pushing or merging commits.

CI helps verify commits if they are compilable.

GitHub CI is easy to setup.

Commit:
Add reminder feature

Commit:
Merge branch 'branch-B-Reminders'

Commit:
Add sort feature

Commit:
Merge branch 'branch-C-Sort'

Commit:
Save task list after being sorted

Commit:
Merge branch 'branch-C-Sort'

Commit:
Add notes feature

Commit:
Merge branch 'branch-D-Notes'

peppapighs committed:
Add launcher class for GUI app

peppapighs committed:
Add user guide for the chatbot

peppapighs committed:
Add window title for the GUI app

peppapighs committed:
Add product screenshot

peppapighs committed:
Improve code quality from soc-se-script

peppapighs committed:
Fix build.gradle for cross-platform build
